### PR TITLE
adding share type circles

### PR DESCRIPTION
**Pull request by ArtificialOwl**
Adding Share_Type_Circles to public auth/backend

### CODE REVIEW

File: apps/dav/lib/Connector/PublicAuth.php [hunk 1]
```diff
@@ -97,7 +97,10 @@ protected function validateUserPass($username, $password) {
 
 		// check if the share is password protected
 		if ($share->getPassword() !== null) {
-			if ($share->getShareType() === \OCP\Share::SHARE_TYPE_LINK || $share->getShareType() === \OCP\Share::SHARE_TYPE_EMAIL) {
+
+			if ($share->getShareType() === \OCP\Share::SHARE_TYPE_LINK
+				|| $share->getShareType() === \OCP\Share::SHARE_TYPE_EMAIL
+				|| $share->getShareType() === \OCP\Share::SHARE_TYPE_CIRCLE) {
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/PublicAuth.php]: Please use the IShare constants

**Review by nickvergessen — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**